20675 cheetah homepage cta radio btns

LanceGundersen opened the pull request:
## Description
As a result of user research, we need to the make the cheetah flow CTA more discoverable.

### Testing instructions
In order to test the project cheetah logic, be sure to disable it `va_online_scheduling_cheetah`.

## Testing done
Local and Unit

## Screenshots
![image](https://user-images.githubusercontent.com/8315447/110552427-5244eb00-8105-11eb-87a9-a88a0876342b.png)
![image](https://user-images.githubusercontent.com/8315447/110552437-596bf900-8105-11eb-8d09-ff41e61d4b5c.png)
![image](https://user-images.githubusercontent.com/8315447/110552487-6daff600-8105-11eb-846d-502b9e4e9155.png)

Non-Refresh Page
<img width="374" alt="image" src="https://user-images.githubusercontent.com/8315447/110656678-27eb4000-818e-11eb-989d-152e408b32bd.png">
<img width="374" alt="image" src="https://user-images.githubusercontent.com/8315447/110656716-35082f00-818e-11eb-96da-26691c4b09a9.png">
<img width="374" alt="image" src="https://user-images.githubusercontent.com/8315447/110656739-3afe1000-818e-11eb-9698-e7befb9f18ea.png">

## Acceptance criteria
- [ ] Changes are behind va_online_scheduling_cheetah feature flag
- [ ] Changes are behind (homepage refresh) feature flag
- [ ] User can select only one of the appointment flows
- [ ] CONTENT - Section heading matches:
Schedule a new appointment
- [ ] CONTENT - Option label copy matches:
Primary or specialty care
COVID-19 vaccine
- [ ] CONTENT - Default CTA label (nothing selected) copy matches:
Choose an appointment type
- [ ] CONTENT - Actionable CTA label copy matches:
Start scheduling
- [ ] If an appointment type is not selected, Schedule appointment CTA is not enabled
- [ ] If an appointment type is selected, Schedule appointment CTA is enabled
- [ ] Given user selects Primary and Specialty care, when user clicks Schedule appointment, Then enter standard appointment flow
- [ ] Given user selects COVID-19 vaccine, when user clicks Schedule appointment, Then enter cheetah appointment flow
- [ ] DESIGN - CTA states match the specs https://adhoc.invisionapp.com/share/URZV3YHCQFZ#/446288321_Action_Links
- [ ] DESIGN - Designs match specs https://adhoc.invisionapp.com/share/URZV3YHCQFZ#/446287949_VAOS-Home-Cta-States-V2

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

LanceGundersen committed:
Add action link styles

- imports action link styles
- adds disabled action link styles

LanceGundersen committed:
Add radio calls to action buttons

- removes old component
- adds UI and logic for radio calls to action
- adds link for starting flow

LanceGundersen committed:
update unit tests

- removes old tests and adds new ones

jbalboni commented:
@LanceGundersen The way I understood Lauren's comment is that this should be behind the project cheetah feature flag, but it should work both on the regular homepage and in the refreshed homepage. Right now, it's only going to show up when the homepage refresh flag is turned on, because it's on the V2 appointment list page.

LanceGundersen committed:
Create ScheduleNewAppointmentRadioButtons.jsx

LanceGundersen committed:
Move radio button to own component

LanceGundersen committed:
Add radio button group component

LanceGundersen committed:
Remove unused components

LanceGundersen committed:
Update unit test

LanceGundersen committed:
Merge branch 'master' into 20675-cheetah-homepage-cta-radio-btns

LanceGundersen committed:
Merge branch 'master' into 20675-cheetah-homepage-cta-radio-btns

vbahinwillit reviewed (commented):
LGTM

LanceGundersen committed:
Fix cypress tests

- Tests broke due to radio selector

LanceGundersen commented on src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:
```diff
@@ -361,113 +362,48 @@ describe('VAOS <AppointmentsPageV2>', () => {
-    ).to.be.ok;
-    expect(screen.getByRole('link', { name: 'Schedule appointment' }));
+
+    expect(await screen.findByRole('link', { name: /Start scheduling/ }));
```
Using the `findByRole` will await the state change after the radio button is clicked.

jbalboni commented on src/applications/vaos/appointment-list/components/AppointmentsPage/ScheduleNewAppointmentRadioButtons.jsx:
```diff
@@ -0,0 +1,74 @@
+        additionalFieldsetClass="vads-u-margin-top--0"
+        onValueChange={({ value }) => {
+          setRadioSelection(value);
+          setRadioSelectionEvent(
```
It looks like you could determine the event name based on the selected value, so you don't necessarily need a second state value here.

Generally, React encourages you to set the minimum amount of state that you need and then derive any additional data based on that when you need it.

LanceGundersen replied:
Removed, good call. I was thinking better to keep it up in a state variable rather than have convoluted ternary operations. Thanks for the feedback 😄

jbalboni commented on src/applications/vaos/appointment-list/components/AppointmentsPage/ScheduleNewAppointmentRadioButtons.jsx:
```diff
@@ -0,0 +1,74 @@
+
+  return (
+    <>
+      <h2 className="vads-u-padding-bottom--0">Schedule a new appointment</h2>
```
This h2 looks too large to me on the current homepage:

![Screen Shot 2021-03-11 at 11 24 51 AM](https://user-images.githubusercontent.com/634932/110819715-7109c580-825c-11eb-8145-9254f92b8be4.png)

LanceGundersen replied:
Good catch, added the h3 class to match the other h2 designs on that page.

LanceGundersen committed:
Update ScheduleNewAppointmentRadioButtons.jsx

- Remove radioSelectionEvent state
- Add JSDoc documentation
- Set showCheetahScheduleButton = false
- Adjust header styling

jbalboni reviewed (commented):
Holding off on reviewing again until changes for the current homepage are made as discussed in Slack.

LanceGundersen committed:
Fix logic for radio cta

- Fix tests
- Fix logic

jbalboni commented on src/applications/vaos/appointment-list/components/AppointmentsPage/ScheduleNewAppointmentRadioButtons.jsx:
```diff
@@ -0,0 +1,89 @@
+ *  showCheetahScheduleButton={valueFromProp}
+ *  startNewAppointmentFlow={givenFlowFromProp()}
+ * />
+ * @module appointment-list/components/ScheduleNewAppointmentRadioButtons
```
I think we should remove `ScheduleNewAppointmentRadioButtons` from the end of this, to avoid cluttering the sidebar of the JSDoc site with every component we might add.

LanceGundersen replied:
Sounds good

jbalboni commented on src/applications/vaos/appointment-list/components/AppointmentsPage/ScheduleNewAppointmentRadioButtons.jsx:
```diff
@@ -0,0 +1,89 @@
+  startNewAppointmentFlow,
+}) {
+  /**
+   * Radio selection
```
Is this comment useful?

LanceGundersen replied:
🤦 That was a JSDoc play I accidentally left in.

jbalboni reviewed (approved):
Couple minor comments, looks good otherwise.

LanceGundersen committed:
Adjust comments

LanceGundersen committed:
Merge branch 'master' into 20675-cheetah-homepage-cta-radio-btns
